src/core/component.js

```diff
@@ -38,11 +40,41 @@ var components = module.exports.components = {};  // Keep track of registered co
  */
 var Component = module.exports.Component = function (el) {
   var rawData = HTMLElement.prototype.getAttribute.call(el, this.name);
+  var tickFn;
+  var self = this;
+
   this.el = el;
   this.data = {};
   this.buildData(this.parse(rawData));
   this.init();
   this.update();
+
+  // Set up tick behavior.
+  if (this.tick !== noopFn) {
+    tickFn = function (t) {
+      // Don't run tick if paused.
+      if (el.paused) { return; }
+      self.tick.bind(self)(t);
+    };
+
+    if (this.isScene) {
+      this.el.addBehavior(tickFn);
+    } else {
+      this.el.sceneEl.addBehavior(tickFn);
+    }
+
+    // Wrap remove to deregister tick on component detach.
+    var removeFn = this.remove;
```

Comment:
Instead of this I would create a removeComponent method on the a-entity and call it wherever we remove a component. I think it's here:

https://github.com/aframevr/aframe/blob/dev/src/core/a-entity.js#L280

and here:

https://github.com/aframevr/aframe/blob/dev/src/core/a-entity.js#L321